iqual-ch/iq_group_bw2
# Issue/us 1289 php81 upgrade

This PR prepares the module to PHP 8 upgrade and add some optimization:

- config for bw2 id required field will be created on module installation
- hooks have been added before and after import, as well as before and after wefborm submission with BW2 handler involved
- custom code specific to gcb hav been removed and will need to be added in custom module, using the new hooks

The module has been tested using the test environment of gcb bw2 instance. It run successfully for the import of users from Drupal to BW2. However the test environment require more parameter than the live and we have currently no documentation on how to populate them, thus posting data of users could not be tested.

Step taken to test the module:

- update iq_group module to latest version ready for PHP 8
- install and configure iq_group_bw2 with GCB Test portal credential from lastpass
- test creating / editing a user: information correctly sent to bw2, but a missing parameter 'Firma' prevent saving the data on their side
- test a webform with IqGroupBW2Submission handler: success
- test an import of the bw2 users into drupal through drush command: success

Additionally the bw2_api module required by this module has been upgrade as well. See PR [here](https://github.com/iqual-ch/bw2_api/pull/1)

Finally the module has been fully tested again on PHP 8 version of the GCB project / iq_group and bw2 and additional required custom code inside hooks.
